File: python/funcionarios/teste.py
from gerente import Gerente
from funcionario import Funcionario
from vendedor import Vendedor
from assistente import Assistente

def folhaSalarial(arrayFuncionarios):
    total = 0
    print('-' * 25)

    for func in arrayFuncionarios:
        try:
            salario_func =  func.calculaSalario()
            total += salario_func

            print(f'{func.nome} || {salario_func}')
            print('-' * 25)
        except:
            continue

    print(f'Folha salarial = {total}')

if __name__ == '__main__':
    n = 7
    lista_func = []
    lista_cpf = []
    lista_nomes = []

    lista_cpf.append("59382082034")
    lista_nomes.append("Lionel Messi")

    lista_cpf.append("47910702086")
    lista_nomes.append("Farid Tari")

    lista_cpf.append("85841264060")
    lista_nomes.append("Cristiano Ronaldo")

    lista_cpf.append("32347464000")
    lista_nomes.append("Lebron James")

    lista_cpf.append("43355267090")
    lista_nomes.append("Lewis Hamilon")

    lista_cpf.append("21198182075")
    lista_nomes.append("Tiger Woods")

    lista_cpf.append("12345678901")
    lista_nomes.append("Nikola Kovac")

    for i in range(n):
        tipo_func = i % 3

        if Funcionario.verificaCPF(lista_cpf[i]):
            match tipo_func:
                case 0:
                    lista_func.append(Gerente(lista_nomes[i], lista_cpf[i]))
                case 1:
                    lista_func.append(Assistente(lista_nomes[i], lista_cpf[i]))
                case 2:
                    lista_func.append(Vendedor(lista_nomes[i], lista_cpf[i], 125.64))
    
    folhaSalarial(lista_func)

File: java/funcionrios/Teste.java
public class Teste {

    public static void calculaFolhaSalarial(Funcionario arr[], int n) {
        System.out.println("--------------------");

        double folhaSalarial = 0;
        for (int i = 0; i < n; i++) {
            try {
                double salarioFunc = arr[i].calculaSalario();
                folhaSalarial += salarioFunc;
    
                System.out.printf("nome: %s | salario: %.2f\n", arr[i].getNome(), salarioFunc);
                System.out.println("--------------------");
            }
            catch (NullPointerException e) {
                continue;
            }
        }

        System.out.printf("Folha salarial: %.2f\n", folhaSalarial);
    }

    public static void main(String args[]) {
        int n = 7;

        Funcionario arrayFunc[] = new Funcionario[n];
        String arrayCPF[] = new String[n];
        String arrayNomes[] = new String[n];

        arrayCPF[0] = "59382082034";
        arrayNomes[0] = "Lionel Messi";

        arrayCPF[1] = "47910702086";
        arrayNomes[1] = "Farid Tari";

        arrayCPF[2] = "85841264060";
        arrayNomes[2] = "Cristiano Ronaldo";

        arrayCPF[3] = "32347464000";
        arrayNomes[3] = "Lebron James";

        arrayCPF[4] = "43355267090";
        arrayNomes[4] = "Lewis Hamilon";

        arrayCPF[5] = "21198182075";
        arrayNomes[5] = "Tiger Woods";

        arrayCPF[6] = "12345678901";
        arrayNomes[6] = "Nikola Kovac";

        for (int i = 0; i < n; i++) {
            int tipoFunc = i % 3;

            if (Funcionario.verificaCPF(arrayCPF[i])) {
                 
                if (tipoFunc == 0)
                    arrayFunc[i] = new Gerente(arrayNomes[i], arrayCPF[i]);

                else if (tipoFunc == 1)
                    arrayFunc[i] = new Assistente(arrayNomes[i], arrayCPF[i]);

                else if (tipoFunc == 2)
                    arrayFunc[i] = new Vendedor(arrayNomes[i], arrayCPF[i], 1202.56);
            }
        }

        Teste.calculaFolhaSalarial(arrayFunc, n);
    }
}
